ludwig/schema/model_types/utils.py

```diff
@@ -337,6 +346,70 @@ def set_llm_tokenizers(config: "ModelConfig") -> None:
+
+    default_max_sequence_length = LLMGenerationConfig().max_new_tokens
+    if config.generation.max_new_tokens != default_max_sequence_length:
+        # Max new tokens is explicitly set by user, so don't override
+        return
+
+    if config.output_features[0].type != TEXT:
+        # This is trickier to set for other output features, so don't override for now.
+        # TODO: Add better support for category output features
+        return
+
+    max_possible_sequence_length = default_max_sequence_length
+    if config.output_features[0].preprocessing.max_sequence_length is not None:
+        # Note: We don't need to check for max between feature.preprocessing.max_sequence_length and
+        # defaults.text.preprocessing.max_sequence_length because the latter is only applied to input features.
+        max_possible_sequence_length = max(
+            default_max_sequence_length, config.output_features[0].preprocessing.max_sequence_length
+        )
+    if config.preprocessing.global_max_sequence_length is not None:
+        # This is not perfect since it includes tokens from both input + output features, but this at least
+        # ensures that max possible of the sequence length is used. It is very likely that the model learns
+        # to generate sequences than this value.
+        max_possible_sequence_length = max(
+            max_possible_sequence_length, config.preprocessing.global_max_sequence_length
+        )
+
+    # It's possible that both max_sequence_length and global_max_sequence_length are not set, in which case
+    # we should fall back to the window size of the pretrained model. By this point, because of schema validation
+    # checks, we know that the base_model exists so we can safely grab the base model's config.
+    # TODO (Arnav): Figure out how to factor in rope scaling factor into this calculation.
+    if max_possible_sequence_length == default_max_sequence_length:
+        model_config = AutoConfig.from_pretrained(config.base_model)
+        max_possible_sequence_length = get_context_len(model_config)
+        # Max length only works if max_new_tokens is not set.
+        # If max_new_tokens is set, then we need to set max_length to None to ensure that the correct number of tokens
+        # are generated (input + output tokens), otherwise we will exceed the bounds of generation resulting in errors.
+        config.generation.max_new_tokens = None
+        config.generation.max_length = max_possible_sequence_length
+        logger.info(
+            f"Setting generation max_length to {max_possible_sequence_length} to correspond with the max sequence "
+            "length assigned to the output feature or the global max sequence length. This will ensure that the "
+            "correct number of tokens are generated at inference time. To override this behavior, set "
+            "`generation.max_length` to a different value in your Ludwig config."
+        )
```

Comment:
Will need to think through this part a bit more, I think the issue I was running into actually happens somewhere else